### PR TITLE
Let integration tests use the CARDANO_WALLET_PORT variable

### PR DESCRIPTION
**Pull request by rvl**
While updating the [Testing](https://github.com/input-output-hk/cardano-wallet/wiki/Testing) wiki page, I thought it would be better if environment variables worked for all tests and benchmarks.

### CODE REVIEW

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs [hunk 1]
```diff
@@ -448,18 +450,17 @@ testMinSeverityFromEnv =
...
-        Right port -> pure $ ListenOnPort port
-        Left e -> die (show e)
+    Just (Right port) -> pure $ ListenOnPort port
+    Just (Left e) -> die $ show e
 
 -- | Directory for extra logging. Buildkite will set this environment variable
 -- and upload logs in it automatically.
 testLogDirFromEnv :: IO (Maybe FilePath)
-testLogDirFromEnv = traverse makeAbsolute =<< lookupEnv "TESTS_LOGDIR"
+testLogDirFromEnv = traverse makeAbsolute =<< lookupEnvNonEmpty "TESTS_LOGDIR"
```
Comment (hasufell) [inline lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:463]: `makeAbsolute ""` equals `pwd`. Isn't that acceptable?

**Review by hasufell — APPROVED** [review]

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs (hunk 1 above)
Comment (rvl) [reply lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:463]: Not really - the normal tests log directory should be used when that variable is empty or unset.

**Comment by rvl** [comment]
Thanks

bors r+

**Comment by rvl** [comment]
bors retry

**Pull request merged**